pkg/secrets/vault_test.go

```diff
@@ -35,3 +40,127 @@ func TestValidateVaultConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestNoSecret(t *testing.T) {
+	notASecret := "notASecret"
+	eng, err := NewDecrypter(context.TODO(), notASecret)
+	assert.Nil(t, err)
+
+	unchanged, _ := eng.Decrypt()
+	assert.Equal(t, notASecret, unchanged)
+}
+
+func TestNoVaultConfig(t *testing.T) {
+	e := Engines["vault"]
+	delete(Engines, "vault")
+	decrypter, err := NewDecrypter(context.TODO(), "encrypted:vault!e:secret!n:test-secret!k:foo")
+	assert.NotNil(t, err)
+	assert.Nil(t, decrypter)
+	Engines["vault"] = e
+}
+
+func TestClientGetsNamespace(t *testing.T) {
+	cases := map[string]struct {
+		cfg VaultConfig
+	}{
+		"with namespace": {
+			cfg: VaultConfig{
+				Enabled:    true,
+				Url:        "http://127.0.0.1:8200",
+				AuthMethod: "KUBERNETES",
+				Namespace:  "my-namespace",
+				Role:       "my-role",
+				Path:       "my-path",
+				Token:      "my-token",
+			},
+		},
+		"no namespace": {
+			cfg: VaultConfig{
+				Enabled:    true,
+				Url:        "http://127.0.0.1:8200",
+				AuthMethod: "KUBERNETES",
+				Namespace:  "",
+				Role:       "my-role",
+				Path:       "my-path",
+				Token:      "my-other-token",
+			},
+		},
+	}
+
+	for _, c := range cases {
+		d := VaultDecrypter{
+			vaultConfig: c.cfg,
+		}
+
+		client, err := d.FetchVaultClient(c.cfg.Token)
+		assert.Nil(t, err)
+
+		header := client.Headers()
+		if header != nil {
+			assert.Equal(t, c.cfg.Namespace, header.Get(consts.NamespaceHeaderName))
+		}
```

Comment:
I think there's a bug in this logic. If something breaks and `SetNamespace` doesn't get called, `header` might always be `nil` and this wouldn't catch it.

If you want to ensure that `SetNamespace` got called if you provide a namespace, you should probably do something like this:

```
if c.cfg.Namespace != "" {
  headers := client.Headers()
  assert.NotNil(t, headers)
  assert.Equal(t, c.cfg.Namespace, headers.get(consts.NamespaceHeaderName)
}
```

Reply:
oh, doh! you're totally right. fixing